src/pages/appointment/host/[appointment_id].tsx

```diff
@@ -0,0 +1,282 @@
+import React, { useContext, useEffect, useState } from "react";
+import Navbar from "@/components/public/Navbar";
+import { Typography, Box, IconButton, Stack, Card, FormHelperText, Grid } from "@mui/material";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { AppointmentDetail } from "@/types/Appointment";
+import { GetAppointmentByAppointmentId, EndAppointment } from "@/services/Appointment";
+import { useRouter } from "next/router";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { Database } from "supabase/db_types";
+import Loading from "@/components/public/Loading";
+import { PAGE_PATHS, ROLE } from "enum/PAGES";
+import { userContext } from "supabase/user_context";
+import TitleTextField from "@/components/post/TitleTextField";
+import LocationTextField from "@/components/post/LocationTextField";
+import DisplayDateTime from "@/components/appointment/DisplayDateTime";
+import Tags from "@/components/post/SelectTags";
+import { Tag } from "@/types/Tag";
+import DescriptionTextField from "@/components/public/DescriptionTextField";
+import DisplayImages from "@/components/post/DisplayImages";
+import { User } from "@/types/User";
+import Participant from "@/components/post/Participant";
+import { COLOR_CODE } from "enum/COLOR";
+import CommonButton from "@/components/public/CommonButton";
+import dayjs from "dayjs";
+
+const HostAppointmentStyle = {
+  TextField: {
+    width: "28vw",
+    minWidth: "400px",
+  },
+  Card: {
+    width: "30vw",
+    minWidth: "450px",
+    paddingTop: "2vh",
+  },
+};
+
+export default function Home() {
+  const router = useRouter();
+  const userStatus = useContext(userContext);
+  const supabaseClient = useSupabaseClient<Database>();
+  const [appointment, setAppointment] = useState<AppointmentDetail | null>();
+  const [isParticipant, setIsParticipant] = useState<boolean | null>(null);
+
+  const appointmentId = parseInt(router.query.appointment_id as string);
+  useEffect(() => {
+    GetAppointmentByAppointmentId(appointmentId, supabaseClient)
+      .then((appointment) => {
+        setAppointment(appointment);
+        setIsParticipant(appointment.ownerId !== userStatus.user?.userId);
+      })
+      .catch((err) => {
+        console.log(err);
+        return;
+      });
+  }, [supabaseClient, appointmentId, userStatus.user]);
+
+  function compareDate(): boolean {
+    if (appointment) {
+      const date: string = appointment.detailHeader.endDateTime.split(" ")[0];
+      const time: string = appointment.detailHeader.endDateTime.split(" ")[1];
+      const am_pm: string = appointment.detailHeader.endDateTime.split(" ")[2];
+
+      const endDate: Date = new Date(
+        Number(date.split("/")[2]),
+        Number(date.split("/")[1]) - 1,
+        Number(date.split("/")[0]),
+        Number(time.split(":")[0]) + Number(am_pm === "PM" ? 12 : 0),
```

Comment:
I think it would cause a problem when time is 12.xx pm or 12.yy am since these times would be 12.xx and 00.yy respectively in 24-hour clock system.
![image](https://user-images.githubusercontent.com/62459518/229300300-bc51aac8-33f6-4012-b855-1f208cd79738.png)

Reply:
As I understand datetime from backend is in xx.xx am/pm (00.00 am as midnight) form and all I did is cast it into 24 hour system. I could not see any issues at this point.

Reply:
If the case you concern is midday (12.00am / 00.00pm), it would be obviously casted as 12.00 in 24 hour

Reply:
So, if it is 12.xx pm it would not cast to 24.xx in 24-hour right?

Reply:
Wouldn't exist tho

Reply:
11.59 pm at maximum, as I inspected ja's response

Reply:
Did you mean user cannot set time as 12.00 pm or it wouldn't cast to 24.xx? Because the datetime picker would allow 12.xx pm so I'm a bit confusing by your word.

Reply:
okay, firstly, the data (datetime rating system get) is from backend system.
![image](https://user-images.githubusercontent.com/69895436/229302567-8d729f1d-53d7-4fdd-9d36-92404fc8b7bc.png)
According to the appended image, It automatically cast as 24-hour.
However, frontend always requires am/pm form (to display), so ja returns datetime like this.
To assume that 00.00 am - 12.00 pm, it would be never happened unless backend possess casting confusion.

Reply:
Ok ka